### PR TITLE
Setup command

### PR DESCRIPTION
**Pull request by FinalAngel**
Adapt setup command to reflect changes and add testing

### CODE REVIEW

File: docs/src/configuration.rst [hunk 1]
```diff
@@ -3,12 +3,47 @@ Configuration
...
+    [tracker]
+    project_id = {gitlab_project_id}
+    group_id = {gitlab_group_id}
+    paused_status =
+    active_status = doing
+    review_status = review
+
+    [timer]
+    project_id = {harvest_project_id}
+    task_id = {harvest_task_id}
```
Comment (GaretJax) [inline docs/src/configuration.rst:61]: This can be greatly simplified by moving the common stuff between projects into the user-level config file.

It might be worth to spend 2 sentences about the locations and inheritance between those files (default-settings.ini < ~/.lancet < project/.lancet)

File: lancet/cli.py [hunk 2]
```diff
@@ -32,7 +33,15 @@
...
-    return sentry_sdk.Client(sentry_dsn, release=__version__)
+    try:
+        return sentry_sdk.Client(sentry_dsn, release=__version__)
+    except sentry_sdk.utils.BadDsn:
+        click.secho(
+            f'Please provide a valid sentry_dsn value in "{USER_CONFIG}"',
+            fg="red",
+            bold=True,
+        )
+        click.secho()
```
Comment (GaretJax) [inline lancet/cli.py]: When does this happen? Might be worth to raise an `Abort` exception instead.

File: lancet/commands/configuration.py [hunk 3]
```diff
@@ -10,60 +10,101 @@
...
+    default=False,
+    help="Adds additional steps to the wizard for debugging.",
+)
 @click.pass_context
-def setup(ctx, force):
-    """Wizard to create the user-level configuration file."""
-    if os.path.exists(USER_CONFIG) and not force:
+def setup(ctx, force, debug):
+    # environment variable used for testing
+    user_path = os.environ.get("ENV_USER_CONFIG") or USER_CONFIG
```
Comment (GaretJax) [inline lancet/commands/configuration.py]: Why the ENV prefix? And this definitely needs a `LANCET` prefix instead.
Also, I'd move it to the settings.py file and make sure that `USER_CONFIG` always reflects the correct value.

File: lancet/commands/configuration.py [hunk 4]
```diff
@@ -10,60 +10,101 @@
...
-                USER_CONFIG
-            ),
+            'An existing configuration file was found at "{}"'.format(user_path),
             fg="red",
             bold=True,
         )
         click.secho(
-            "Please remove it before in order to run the setup wizard or use\n"
-            "the --force flag to overwrite it."
+            "Please remove it before in order to run the setup wizard or use"
```
Comment (GaretJax) [inline lancet/commands/configuration.py]: The line breaks in the python code don't have to reflect the ones in the terminal. I'd undo this change.

File: lancet/commands/configuration.py [hunk 5]
```diff
@@ -89,7 +130,8 @@ def init(ctx, force):
         )
         ctx.exit(1)
 
-    project_key = click.prompt("Project key on the issue tracker")
+    project_id = click.prompt("Project id on Gitlab")
```
Comment (GaretJax) [inline lancet/commands/configuration.py:157]: Lancet supports gitlab and jira, avoid hardcoding specific services.

File: lancet/commands/configuration.py [hunk 6]
```diff
@@ -89,7 +130,8 @@ def init(ctx, force):
...
+    project_group = click.prompt("Project group on Gitlab")
```
Comment (GaretJax) [inline lancet/commands/configuration.py:158]: Ditto.

File: setup.cfg [hunk 7]
```diff
@@ -1,2 +1,5 @@
 [aliases]
 test=pytest
+
+[tool:pytest]
+addopts = -p no:warnings
```
Comment (GaretJax) [inline setup.cfg]: Why not? What warnings are there?

**Review by GaretJax — CHANGES_REQUESTED** [review]

File: docs/src/configuration.rst (hunk 1 above)
Comment (FinalAngel) [reply docs/src/configuration.rst:61]: adapted docs accordingly

File: lancet/cli.py (hunk 2 above)
Comment (FinalAngel) [reply lancet/cli.py]: switch to exit with proper text, seems more clean and added comments :)

File: lancet/commands/configuration.py (hunk 3 above)
Comment (FinalAngel) [reply lancet/commands/configuration.py]: Switched to specific lancet_test prefix. I tried moving it into the settings.py as it seemed better to me. The problem is that the env variable is loaded inside the setup function and everything outside (settings) are not updated. I wasn't able to get over that obstacle

File: lancet/commands/configuration.py (hunk 4 above)
Comment (FinalAngel) [reply lancet/commands/configuration.py]: reverted

File: lancet/commands/configuration.py (hunk 5 above)
Comment (FinalAngel) [reply lancet/commands/configuration.py:157]: Done

File: setup.cfg (hunk 7 above)
Comment (FinalAngel) [reply setup.cfg]: I added it cause the following warning shows up originating from the jinja2 template language:

```
/lancet-dev/lib/python3.7/site-packages/jinja2/utils.py:485
/lancet-dev/lib/python3.7/site-packages/jinja2/utils.py:485: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import MutableMapping

/lancet-dev/lib/python3.7/site-packages/jinja2/runtime.py:318
/lancet-dev/lib/python3.7/site-packages/jinja2/runtime.py:318: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Mapping

-- Docs: https://docs.pytest.org/en/latest/warnings.html
```

Originating from https://github.com/pallets/jinja/pull/867

File: lancet/cli.py (hunk 2 above)
Comment (FinalAngel) [reply lancet/cli.py]: Had to revert cause aborting prevents the file to be rewritten using lancet setup -f. The warning should show it a command is run and sentry_dsn is incorrectly configured

File: lancet/commands/configuration.py (hunk 6 above)
Comment (FinalAngel) [reply lancet/commands/configuration.py:158]: Ditto :)

File: lancet/commands/configuration.py (hunk 5 above)
Comment (FinalAngel) [reply lancet/commands/configuration.py:157]: However, I need to check the generations for gitlab and jira once checking and adapting these commands. The scr-manager probably also need to be updated depending on the choice. For now the important one is the test to check the expected outcome. I expect the setup command to evolve the deeper we dig.

File: setup.cfg (hunk 7 above)
Comment (FinalAngel) [reply setup.cfg]: Also added that as a comment

Comment (GaretJax) [reply setup.cfg]: I'd rather not hide warnings. You should be able to disable that specific warning (including the specific location).

File: lancet/cli.py (hunk 2 above)
Comment (GaretJax) [reply lancet/cli.py]: Why would it prevent it? Also, use an exception instead of sys.exit; easier to handle and easier to test for.

File: lancet/commands/configuration.py [hunk 8]
```diff
@@ -102,15 +168,22 @@ def init(ctx, force):
     venv_path = click.prompt("Path to virtual environment", default=venv)
 
     project_id = click.prompt("Project ID on Harvest", type=int)
-    task_id = click.prompt("Task id on Harvest", type=int)
+    task_id = click.prompt("Task ID on Harvest", type=int)
```
Comment (GaretJax) [inline lancet/commands/configuration.py]: Same above for project *id* on gitlab :P

**Review by GaretJax — CHANGES_REQUESTED** [review]

File: lancet/cli.py (hunk 2 above)
Comment (FinalAngel) [reply lancet/cli.py]: @GaretJax any command that runs, if there is an incorrect sentry_dsn added will fail, even if I want to re-setup it again using ``lancet setup -f`` resulting into editing the file manually to fix the issue if I hard abort (exit, or raise exception). That's why I rather have it as a Warning rather than aborting the command.

File: setup.cfg (hunk 7 above)
Comment (FinalAngel) [reply setup.cfg]: Adapted :)
